### ISSUE TITLE
Fix SSL errors with local dev server

### ISSUE THREAD

**Issue by enoren5** [enoren5/tarot_juicer#228]
When running the server in my local development environment, it throws an SSL error. Here is the traceback:
```
$ python manage.py runserver 8000

Secure Mode Enabled: DEBUG MODE IS FALSE

Database Config: {'default': {'NAME': '/home/<user>/dev/projects/python/2018-and-2020/tarot_juicer/db.sqlite3', 'USER': '', 'PASSWORD': '', 'HOST': '', 'PORT': '', 'CONN_MAX_AGE': 600, 'ENGINE': 'django.db.backends.sqlite3'}}

Performing system checks...

System check identified no issues (0 silenced).
January 22, 2023 - 11:09:37
Django version 3.2.14, using settings 'tarot_juicer.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
Ist Print
[22/Jan/2023 11:09:38] code 400, message Bad request version ('ÊÊ\x13\x01\x13\x02\x13\x03À+À/À,À0Ì©Ì¨À\x13À\x14\x00\x9c\x00\x9d\x00/\x005\x01\x00\x01\x93ZZ\x00\x00\x00\x17\x00\x00ÿ\x01\x00\x01\x00\x00')
[22/Jan/2023 11:09:38] You're accessing the development server over HTTPS, but it only supports HTTP.

[22/Jan/2023 11:09:38] code 400, message Bad request version ('êê\x13\x01\x13\x02\x13\x03À+À/À,À0Ì©Ì¨À\x13À\x14\x00\x9c\x00\x9d\x00/\x005\x01\x00\x01\x93ªª\x00\x00\x00\x17\x00\x00ÿ\x01\x00\x01\x00\x00')
[22/Jan/2023 11:09:38] You're accessing the development server over HTTPS, but it only supports HTTP.
```
Even when I specifically enter: `http://127.0.0.1:8000/` (without the `s` after `http`) the issue is still present. Django still doesn’t serve the web page. 

I discovered a work around which was supposed to be temporary but I have been using for a long time. I documented it in the README.md under **Usage Notes** here: “[#7. Local dev server over HTTPS](https://github.com/enoren5/tarot_juicer#7-local-dev-server-over-https)”. If you in turn follow the guide referred to in the README (“[How to run a local Django development server over HTTPS with a trusted self-signed SSL certificate](https://timonweb.com/django/https-django-development-server-ssl-certificate/)”), which involves setting up SSL certificates locally, you can run the local Django dev server using: `$ python manage.py runserver_plus --cert-file cert.pem --key-file key.pem`.  While this workaround works for me, it should no longer be necessary. Your task is to get Django to run using the simple way with just: `$ python manage.py runserver 8000`.

**Comment by PanchalDeveloper** [comment]
Hi Enoren!

I'm Aman, A full-stack webdeveloper  working as a freelancer on Fiverr and Upwork.
I saw you request on Upwork for "Django CMS : Correct SSL error + Sessions/Authentifcation code refactor".

1. About your issue with running the Django server, Have you tried running the Django server with "python manage.py runserver 0.0.0.0:8000" or "python3 manage.py runserver 0.0.0.0:8000" commands ?

Alternatively, you can use a reverse proxy like Nginx or Apache to handle the HTTPS connection and forward the requests to the Django development server via HTTP. This is a more secure and production-ready approach, but it does require some additional setup and configuration, Or if you want I can do it for you.

2. About "Major Code Refactor" issue: 

To replace a custom "password protect" middleware with the official Django session and authentication middleware system, you will need to make a few changes to your code like removing your custom middleware from your Django project's settings, 'MIDDLEWARE' setting or 'MIDDLEWARE_CLASSES' if you are using older version of django etc.

Please reachout to me on Upwork (https://www.upwork.com/freelancers/~01c956f7c6987fd453) if you are intrested.

Regards,
Aman

**Comment by enoren5** [comment]
Documented and resolved in PR #236
